src/main/java/com/solvd/laba/utils/WebDriverFactory.java

```diff
@@ -13,29 +18,49 @@ public class WebDriverFactory {
     private static final ThreadLocal<WebDriver> driver = new ThreadLocal<>();
     private static final Logger logger = LoggerFactory.getLogger(WebDriverFactory.class);
 
-    public static WebDriver getDriver() {
-        if (driver.get() == null) {
-            EnvironmentConfig environmentConfig = new EnvironmentConfig();
-            String browser = environmentConfig.getProperty("browser", "chrome");
-
-            switch (browser.toLowerCase()) {
-                case "chrome":
-                    WebDriverManager.chromedriver().setup();
-                    driver.set(new ChromeDriver());
-                    logger.info("Launched Chrome browser");
-                    break;
-                case "firefox":
-                    WebDriverManager.firefoxdriver().setup();
-                    driver.set(new FirefoxDriver());
-                    logger.info("Launched Firefox browser");
-                    break;
-                default:
-                    throw new RuntimeException("Browser not supported: " + browser);
+    public static WebDriver getDriver(String browser) {
+        EnvironmentConfig config = new EnvironmentConfig();
+        String runMode = config.getProperty("run.mode", "local").toLowerCase();
+
+        try {
+            if (runMode.equals("grid")) {
```

Comment:
Set up Selenium Grid and use it to execute tests in parallel mode using that grid
the "run.mode" switch between local and grid is not necessary. Your entire focus is on learning and using Selenium Grid, so the code should always run using RemoteWebDriver against the Grid.